SoftLayer/managers/vs.py

```diff
@@ -1002,6 +1003,25 @@ def _get_price_id_for_upgrade_option(self, upgrade_prices, option, value, public
                 else:
                     return price.get('id')
 
+    def get_summary_data_usage(self, instance_id, start_date=None, end_date=None, valid_type=None, summary_period=None):
+        """Retrieve the usage information of a virtual server.
+
+        :param string instance_id: a string identifier used to resolve ids
+        :param string start_date: the start data to retrieve the vs usage information
+        :param string end_date: the start data to retrieve the vs usage information
+        :param string string valid_type: the Metric_Data_Type keyName.
+        :param int summary_period: summary period.
+        """
+        valid_types = [
+            {
+                "keyName": valid_type,
+                "summaryType": "max"
+            }
+        ]
+
+        return self.metric_tracking_object.getSummaryData(start_date, end_date, valid_types, summary_period,
```

Comment:
getSummaryData is likely to return a lot of results if the time period is long enough, so we should be using pagination here. Simply changing the call to

`return self.client.call('Metric_Tracking_Object', 'getSummaryData', start_date, end_date, valid_types, summary_period, id=instance_id, iter=True)` should work here.

But let me know if you have any questions about how that works.

Reply:
Done.